ckanext/switzerland/plugin.py

```diff
@@ -227,6 +232,70 @@ def _package_map_ckan_default_fields(self, pkg_dict):  # noqa
 
         return pkg_dict
 
+    def _prepare_resources_format(self, pkg_dict):
+        if pkg_dict.get('resources') is not None:
+            for resource in pkg_dict['resources']:
+                resource = self._prepare_resource_format(resource)
+
+        return pkg_dict
+
+    # Generates format of resource and saves it in format field
+    def _prepare_resource_format(self, resource):
+        resource_format = ''
+
+        # get format from download_url file extension if available
+        if 'download_url' in resource and resource['download_url']:
+            url = resource['download_url']
+            path = urlparse.urlparse(url).path
+            ext = os.path.splitext(path)[1]
+            if ext:
+                resource_format = ext.replace('.', '').lower()
+                resource['format'] = self._map_to_valid_format(resource_format)
+                return resource
+
+        # get format from media_type field if available
+        if 'media_type' in resource and resource['media_type']:
```

Comment:
use `if not resource_format and and resource.get('media_type') is not None:` and get rid of the return statement